pkg/istio/rbac/v2/provider.go

```diff
@@ -253,13 +289,44 @@ func (p *v2) ConvertAthenzModelIntoIstioRbac(athenzModel athenz.Model, serviceNa
 			log.Errorln("error when convert role to spiffe name: ", err.Error())
 			continue
 		}
-		from_principal.Source.Principals = append(from_principal.Source.Principals, roleSpiffeName)
-		rule.From = append(rule.From, from_principal)
-		if len(from_namespace.Source.Namespaces) > 0 {
-			rule.From = append(rule.From, from_namespace)
+		if combinationPolicyFlag {
+			from_principalAndRequestPrincipal.Source.Principals = append(from_principalAndRequestPrincipal.Source.Principals, roleSpiffeName)
+			from_principalAndNotRequestPrincipal.Source.Principals = append(from_principalAndNotRequestPrincipal.Source.Principals, roleSpiffeName)
+			for _, proxyPrincipal := range proxyPrincipalsList {
+				proxySpiffeName, err := common.MemberToSpiffe(proxyPrincipal)
+				if err != nil {
+					log.Errorln("error converting role member to spiffeName: ", err.Error())
+					continue
+				}
+				if p.enableOriginJwtSubject {
+					originJwtName, err := common.MemberToOriginJwtSubject(proxyPrincipal)
+					if err != nil {
+						log.Errorln(err.Error())
+						continue
+					}
+					from_principalAndRequestPrincipal.Source.RequestPrincipals = append(from_principalAndRequestPrincipal.Source.RequestPrincipals, originJwtName)
+				}
+				from_principalAndRequestPrincipal.Source.Principals = append(from_principalAndRequestPrincipal.Source.Principals, proxySpiffeName)
+			}
+			from_principalAndNotRequestPrincipal.Source.NotRequestPrincipals = append(from_principalAndNotRequestPrincipal.Source.NotRequestPrincipals, "*")
+		} else {
+			from_principal.Source.Principals = append(from_principal.Source.Principals, roleSpiffeName)
 		}
-		if p.enableOriginJwtSubject && len(from_requestPrincipal.Source.RequestPrincipals) > 0 {
-			rule.From = append(rule.From, from_requestPrincipal)
+
+		if combinationPolicyFlag {
+			rule.From = append(rule.From, from_principalAndRequestPrincipal)
+			rule.From = append(rule.From, from_principalAndNotRequestPrincipal)
+			if len(from_namespace.Source.Namespaces) > 0 {
```

Comment:
Should we move the namespace check / append out of the if statements since it's the same for both?